tests/testthat/test-visualizations.R

```diff
@@ -0,0 +1,92 @@
+  expect_error(plotCellVsReference(test,1,training,1), NULL) # looks for logcounts, but the dataset does not have logcounts
+  expect_s3_class(P <- plotCellVsReference(test, 1, training, 1, 1, 1), "ggplot")
+  expect_s3_class(plotCellVsReference(test.exp, 1, training.exp, 1), "ggplot")
+  expect_s3_class(plotCellVsReference(test, 1, training.exp, 1,1,NULL), "ggplot")
+  expect_s3_class(plotCellVsReference(test.exp, 1, training, 1,NULL,1), "ggplot")
+  expect_false(isTRUE(all.equal(P,
+                                P2 <- plotCellVsReference(test, 2, training, 1, 1, 1))))
+  expect_false(isTRUE(all.equal(P, 
+                                P3 <- plotCellVsReference(test, 1, training, 2, 1, 1))))
+  expect_false(isTRUE(all.equal(P2, P3)))
+})
+
+example("SingleR")
+colnames(test) <- paste0("name",seq_len(ncol(counts(test))))
+pred <- SingleR(test, sce, labels=sce$label)
+
+test_that("We can produce heatmaps of scores with plotScoreHeatmap", {
+  expect_s3_class(plotScoreHeatmap(results = pred),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                          cells.use = 1:50),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                          cells.use = rownames(pred)[1:50]),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   labels.use = levels(as.factor(pred$labels))[1:3]),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   clusters = pred$labels),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   max.labels = length(levels(as.factor(pred$labels)))-1),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   normalize = FALSE),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   cells.order=seq_len(nrow(pred))),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   clusters = pred$labels,
+                                   order.by.clusters=TRUE),
+                  "pheatmap")
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   silent=TRUE),
+                  "pheatmap")
+})
+
+test_that("Explicit visualization adjustments don't break plotScoreHeatmap plotting.",{
+  expect_s3_class(plotScoreHeatmap(results = pred,
+                                   fontsize.row = 5),
+                  "pheatmap")
+})
+
+test_that("We can pass excess pheatmap::pheatmap parameters through plotScoreHeatmap.", {
+  plotScoreHeatmap(results = pred,
+                   cutree_col = 3)
+})
+
+# I'm not sure how best to compare two plots (ggplot nor pheatmap).
```

Comment:
I'm not sure how best to compare two plots
This part is a backbone for checking if certain inputs would produce the same plot when they should.  But I'm not actually sure the proper test_code to use.

Reply:
It's fine to not have to compare plots; the main aim is to ensure that they run.

(Technically, you could compare _ggplot_ objects. But this is overkill in most applications.)

FYI, `test_that` chunks seem to be skipped if there is not at least one `expect_*` statement inside them. Even if you don't compare anything, you should do something like

```r
expect_error(run_cmd_here(), NA)
```

... to indicate that a command should run _without_ any error.

Reply:
Fixed.
Since it is not necessary, I've removed the p[lot comparisons.
I also added an expect_* statement in the spot I'd previously forgotten too.